Base Search Bar Done

myles-block opened the pull request:
Demo of Base Search Bar:

https://user-images.githubusercontent.com/85708958/183532849-323e04b6-a193-467d-a314-36fcce1ff085.mov


-Finished base search bar

myles-block committed:
Added & Intialized Search Bar w/ dummy values

myles-block committed:
Working on Search Bar...searchResultsare nil

Add Object isn't working properly for resultNames, not adding the product name given from search (even though they exist in the NSLog)

myles-block committed:
Tried Debugging on nil search

myles-block committed:
Search Bark work w/ one keyword

myles-block committed:
stopped on search, switch to Details View fix

myles-block committed:
paused on Details View, switched back to Search

myles-block committed:
Clean Up Code

laithalq commented on Orion/Orion/API/APIManager.m:
```diff
@@ -80,11 +80,36 @@ - (void)getCategorySpecificTrending:(NSDictionary *)passedCategoryInfo completio
+               NSDictionary *dataDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
+               NSArray *resultsArray = dataDictionary[@"products"];//results array of trending products
+               NSMutableArray *searchProducts = [Product productsWithProductAPIArray:resultsArray];
+               NSLog(@"%@", @"Logged✅✅✅");
```
For style purposes lets remove the extra logs!

laithalq commented on Orion/Orion/API/APIManager.m:
```diff
@@ -80,11 +80,36 @@ - (void)getCategorySpecificTrending:(NSDictionary *)passedCategoryInfo completio
 }
 
+- (void)getSearching:(NSString *)passedSearchWord completion:(void(^)(NSArray *products))completion {
+    NSString *searchParam = [NSString stringWithFormat:@"%@%@%@%@%@", productAPIURLString, @"(name=", passedSearchWord, @"*)?format=json&show=sku,name,regularPrice,image&pageSize=10&apiKey=" ,self.api_key];
```
since the list of products comes back in pages, we should support pulling the next page when the user reaches the end of the list! Otherwise they'll only ever see the top 10 results for the search

laithalq commented on Orion/Orion/Models/Product.h:
```diff
@@ -17,12 +17,16 @@ NS_ASSUME_NONNULL_BEGIN
 //@property CGFloat *productPrice;
 
-
+//@property BOOL *productAPICheck;//checks if intialized from productAPI
```
style: please remove commented out code/properties

laithalq commented on Orion/Orion/Models/Product.m:
```diff
@@ -7,37 +7,89 @@
+        self.productPrice = dictionary[@"prices"][@"regular"];
     }
     
+//    if(productAPICheck == false)//initializes based on which API is being used
```
nit: commented out code

laithalq commented on Orion/Orion/Models/Product.m:
```diff
@@ -7,37 +7,89 @@
     return self;
 }
 
+//- (instancetype)addedWithProductSpec:(Product *)dictionary {
```
nit: commented out code

laithalq commented on Orion/Orion/Models/Product.m:
```diff
@@ -7,37 +7,89 @@
+    return products;
+}
+
+//+ (Product *)detailProductAdditions: (Product *)sentProduct {
```
ditto, to avoid spam i'll stop mentioning these here. But good to do a quick scan over "//" and"/*" instances and clear out non-descriptive comments

laithalq reviewed (commented):
Left some comments! Some on style (mainly commented out code and leftover logs) and one main comment around pagination and pulling more than just the first page of results (ie: when the user reaches the bottom of the list, firing off a query for the next page)

laithalq commented:
Here's an example of how to check for reaching the bottom of the tableview and from there you'd trigger a query for the next page (will probably need to maintain your own page counter that resets when you start a new search) https://stackoverflow.com/questions/39015228/detect-when-uitableview-has-scrolled-to-the-bottom

myles-block committed:
Fixed Details View & Fixed PFP

myles-block committed:
Fixed Misc Features + Back4App Product Save

Fixed/Added a lot of missing features (pfp works, details works, bookmarks almost works)...added custom parse model to server

myles-block committed:
Continued working on Back4App save in Background

myles-block committed:
Bookmark works + Added Logo

myles-block committed:
Fixed some Bookmark Bugs...prepping for PR!
